astropy/nddata/nddata.py

```diff
@@ -5,170 +5,239 @@
+        Defaults to ``None``.
+
+    wcs : any type, optional
+        A world coordinate system (WCS) for the dataset.
+        Default is ``None``.
 
-    mask : `~numpy.ndarray`-like, optional
-        Mask for the data. The values must be ``False`` where
-        the data is *valid* and ``True`` when it is not (like Numpy
-        masked arrays). If ``data`` is a numpy masked array, providing
-        ``mask`` here will causes the mask from the masked array to be
-        ignored.
+    meta : `dict`-like object, optional
+        Meta information about the dataset. If no meta is provided an empty
+        `collections.OrderedDict` is created.
 
-    wcs : undefined, optional
-        WCS-object containing the world coordinate system for the data.
+    unit : `~astropy.units.Unit`-like or str, optional
+        Unit for the dataset.
+        Default is ``None``.
 
-    meta : `dict`-like object, optional
-        Metadata for this object. Must be dict-like but no further restriction
-        is placed on meta.
+    copy : `bool`, optional
+        Save the attributes as copy or as reference. ``True`` copies every
+        attribute before saving it while ``False`` tries to save every
+        parameter as reference. Note however that it is not always possible to
+        save each input as reference.
+        Default is ``False``.
 
-    unit : `~astropy.units.UnitBase` instance or str, optional
-        The units of the data. If data is an `~astropy.units.Quantity` then
-        ``unit`` is set to the unit of the data; is a unit is also explicitly
-        provided an error is raised.
+    Raises
+    ------
+    TypeError:
+        In case any parameter does not fulfill the classes restrictions.
 
     Notes
     -----
-    The data in a `NDData` object should be accessed through the data
+    Each attribute can be accessed through the homonymous instance attribute,
+    such as data in a `NDData` object can be accessed through the ``data``
     attribute.
 
     For example::
 
         >>> from astropy.nddata import NDData
-        >>> x = NDData([1,2,3])
-        >>> x.data
+        >>> nd = NDData([1,2,3])
+        >>> nd.data
         array([1, 2, 3])
+
+    Given an implicit parameter and an explicit one during initialization, for
+    example the ``data`` is a `~astropy.units.Quantity` and the unit parameter
+    is not None. Then the implicit parameter is replaced (without conversion)
+    by the explicit one and a warning is issued::
+
+        >>> import numpy as np
+        >>> import astropy.units as u
+        >>> q = np.array([1,2,3,4]) * u.m
+        >>> nd2 = NDData(q, unit=u.cm)  # doctest: +SKIP
+        INFO: Overwriting Quantity's current unit with specified unit [astropy.nddata.nddata]
+        >>> nd2.data  # doctest: +SKIP
+        array([ 1.,  2.,  3.,  4.])
+        >>> nd2.unit  # doctest: +SKIP
+        Unit("cm")
+
+    See also
+    --------
+
+    - `~astropy.nddata.mixins.NDIOMixin`
```

Comment:
Nice, I never thought that sphinx automatically parses the see also part as links. Thanks for the help!